pkg/internal/local/klient/kind.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package klient
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cloudoperators/greenhouse/pkg/internal/local/utils"
+)
+
+// CreateCluster - creates a kind cluster with the given name
+// if the cluster already exists, it sets the context to the existing cluster
+func CreateCluster(clusterName string) error {
+	exists, err := clusterExists(clusterName)
+	if err != nil {
+		return err
+	} else if exists {
+		utils.Logf("kind cluster with name %s already exists", clusterName)
+		return utils.Shell{
+			Cmd: "kubectl config set-context kind-${name}",
+			Vars: map[string]string{
+				"name": clusterName,
+			},
+		}.Exec()
+	}
+	return utils.Shell{
+		Cmd: "kind create cluster --name ${name}",
```

Comment:
I'm not a big fan of calling another CLI like this. 
Could we not use something like this to trigger the cluster creation from go code ?
https://pkg.go.dev/sigs.k8s.io/e2e-framework/support/kind@v0.4.0#Cluster.CreateWithConfig

Reply:
Sure, please make sure that we don't forget it.